src/org/rascalmpl/library/lang/paths/Windows.rsc

```diff
@@ -0,0 +1,113 @@
+as well as shares on local networks (UNC notation). It also derives from openly available documentation 
+sources on Windows and the .NET platform for confirmation and test examples.
+
+The main function of this module, ((parseWindowsPath)):
+* faithfully maps any syntactically correctly Windows paths to syntactically correct `loc` values.
+* throws a ParseError if the path does not comply. Typically file names ending in spaces do not comply.
+* ensures that if the file exists on system A, then the `loc` representation
+resolves to the same file on system A via any ((module::IO)) function. 
+* and nothing more. No normalization, no interpretatioon of `.` and `..`, no changing of cases. 
+This is left to downstream processors of `loc` values, if necessary. The current transformation
+is purely syntactical, and tries to preserve the semantics of the path as much as possible.
+}
+module lang::paths::Windows
+
+import IO;
+import util::SystemAPI;
+
+lexical WindowsPath
+    = unc              : Slash Slash Slashes? PathChar* hostName Slashes PathChar* shareName Slashes WindowsFilePath path
+    | absolute         : Drive drive ":" Slashes WindowsFilePath path
+    | driveRelative    : Drive drive ":" WindowsFilePath path
+    | directoryRelative: Slash WindowsFilePath
+    | relative         : WindowsFilePath path         
+    ;
+
+lexical PathChar = !([\a00-\a20\< \> : \" | ? * \\ /] - [\ ]);
+
+lexical PathSegment
+    = current: "."
+    | parent : ".."
+    | name   : PathChar+ \ ".." \ "."
+    ;
+
+lexical Drive = [A-Za-z];
+
+lexical Slashes = Slash+ !>> [\\/];
+
+lexical Slash = [\\/];
+
+lexical WindowsFilePath = {PathSegment Slashes}* segments Slashes? [\ .] !<< (); // only the last segment must not end in spaces.
+
+import ParseTree;
+
+@synopsis{Convert a windows path literal to a source location URI}
+@description{
+1. parses the path using the grammar for ((WindowsPath))
+2. takes the _literal_ name components using string interpolation `"<segment>"`. This means no decoding/encoding happens at all while extracting
+hostname, share name and path segment names. Also all superfluous path separators are skipped.
+3. uses `loc + str` path concatenation with its builtin character encoding to construct the URI. Also
+the right path separators are introduced. 
+}
+loc parseWindowsPath(str input, loc src=|unknown:///|) = mapPathToLoc(parse(#WindowsPath, input, src));
+
+@synopsis{UNC}
+loc mapPathToLoc((WindowsPath) `<Slash _><Slash _><Slashes? _><PathChar* hostName><Slashes _><PathChar* shareName><Slashes _><WindowsFilePath path>`)
+    = (|unc://<hostName>/| + "<shareName>" | it + "<segment>" | segment <- path.segments );
+
+@synopsis{Absolute: given the drive and relative to its root.}
+loc mapPathToLoc((WindowsPath) `<Drive drive>:<Slashes _><WindowsFilePath path>`) 
+    = (|file:///<drive>:/| | it + "<segment>" | segment <- path.segments);
```

Comment:
I thought we needed to make it into a string first? so `file:///<"<drive>">:/` ?

Reply:
No that's idempotent. Both do string interpolation. The problem is with string interpolation in the path part of a loc; then the encoding feature is broken. So I'm using `+` for the path part.